packages/jetpack-ai-calypso/src/logo-generator/components/logo-presenter.tsx

```diff
@@ -10,32 +10,62 @@ import { useState } from 'react';
  */
 import LogoIcon from '../assets/icons/logo';
 import MediaIcon from '../assets/icons/media';
+import useLogo from '../hooks/use-logo';
+import { saveToMediaLibrary } from '../lib/save-to-media-library';
+import { ImageLoader } from './image-loader';
 import './logo-presenter.scss';
 /**
  * Types
  */
-import { Logo } from '../store/types';
-import { ImageLoader } from './image-loader';
+import type { Logo } from '../store/types';
 import type React from 'react';
 
 const SaveInLibraryButton: React.FC = () => {
 	const [ saved, setSaved ] = useState( false );
+	const [ saving, setSaving ] = useState( false );
+
+	const {
+		selectedLogo,
+		site: { id },
+	} = useLogo();
 
 	return (
 		<button
 			className={ classnames( 'jetpack-ai-logo-generator-modal-presenter__action', {
-				clickable: ! saved,
+				clickable: ! saved && ! saving,
 			} ) }
-			onClick={ () => {
-				if ( ! saved ) {
-					setSaved( true );
+			onClick={ async () => {
+				if ( ! saving && ! saved ) {
+					try {
+						setSaving( true );
+
```

Comment:
Let's start adding track events early on or we'll forget about them.

Reply:
https://github.com/Automattic/wp-calypso/issues/85858
We created an issue for this.